app/helpers/api_page_helper.rb

```diff
@@ -0,0 +1,50 @@
+module ApiPageHelper
+  # ask for current_weather
+  def get_weather(city = 'berlin', api_id)
+    # Rails.cache.clear
+    if api_id == 1
+      cache_name = 'curr_weather'
+    elsif api_id == 2
+      cache_name = 'forecast_weather'
+    end
+
+    exists_weather_of(city, api_id, cache_name)
+    if api_id == 1
+      get_curr_weather_data(Rails.cache.fetch('curr_weather'))
```

Comment:
I'm not sure exactly what's being cached, but if it's the current weather for a specific city, you should probably use a specific key name like `Rails.cache.fetch("curr_weather_for_#{city}")`. If you have an ID, that would be a better variable to use. The problem is there are common city names throughout the world (There is another Berlin in the US). 

So, if you think you don't need specific weather data for different cities cached separately, then you can keep overriding the same key. You have to be careful though because `curr_weather` data might be data from Berlin and if someone looks for the weather in Frankfurt, make sure you're not reading the cached information for Berlin.